### ISSUE TITLE
[bigquery-io] - Support for `offset` in ReadRowsRequest

### ISSUE THREAD

**Issue by nmatare** [tensorflow/io#876]
The v2 BigQuery storage API [apiv1beta2](https://cloud.google.com/bigquery/docs/reference/storage/rpc/google.cloud.bigquery.storage.v1beta2#google.cloud.bigquery.storage.v1beta2.ReadRowsRequest) introduces an `offset` parameter which may be passed to select the starting position of the stream. While I understand the current implementation uses the beta1 API, it would be great to introduce this into the `tfio.bigquery.BigQueryReadSession` so as to start reading the underlying table at a prespecified index.

https://github.com/tensorflow/io/blob/ce58581f71778b3a797586fa1228a399c7b3a167/tensorflow_io/bigquery/kernels/bigquery_kernels.cc#L139

Currently, our workflow looks something like:
```
OFFSET = 1000
session = tf_io_bigquery_client.read_session()

dataset = (tf.data.Dataset
  .from_tensor_slices(session._streams)
  .flat_map(session.read_rows)
  .skip(OFFSET)
)
```
However, this requires us to iterate through the entire stream to skip `OFFSET` number of rows which naturally takes more time and incurres billing costs.

**Comment by sshrdp** [comment]
Do you create a single stream? Because the offset applies to each stream read operation and having a single offset applied to multiple streams doesn't make a lot of sense. 
Is the goal to do sharding for distributed training?

**Comment by nmatare** [comment]
Yup, exactly; we're only creating one stream (requested_streams=1) and then using flat_map to read the data sequentially. 

For each worker, we randomly select a starting position to begin reading the underlying table. We repeat this for each epoch -- randomly selecting the starting the position on each iteration for each worker.

The google.cloud.bigquery.storage docs appear to read that you can use multiple streams while still reading the data sequentially, but I'm unsure of this so we've just stuck to using one stream per each worker.

**Comment by vlasenkoalexey** [comment]
Added code to pass through offset to BQ Storage API, also it does not return any data when offset is not 0. Filed internal bug for BQ team to address that (b/153579992).
It should work, BQ is updated.
